src/quo/components/wallet/confirmation_progress/view.cljs

```diff
@@ -0,0 +1,82 @@
+(ns quo.components.wallet.confirmation-progress.view
+  (:require [quo.components.wallet.confirmation-progress.style :as style]
+            [quo.components.wallet.progress-bar.view :as progress-box]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]))
+
+(defn- calculate-box-state
+  [state counter index]
+  (cond
+    (and (= state :sending) (>= counter index) (< index 3))                 :confirmed
+    (and (= state :confirmed) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalising) (>= counter index) (< index 5))              :confirmed
+    (and (= state :finalising) (>= counter index) (> index 4) (< index 20)) :finalized
+    (and (= state :finalized) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalized) (>= counter index) (> index 4))               :finalized
+    (and (= state :error) (>= counter index) (< index 2))                   :error
+    :else                                                                   :pending))
+
+(defn- calculate-box-state-network-left
+  [state network]
+  (cond
+    (= state :error)                                                     :error
+    (and (= network :arbitrum) (= state :sending))                       :confirmed
+    (or (= state :confirmed) (= state :finalising) (= state :finalized)) :confirmed
+    :else                                                                :pending))
+
+(defn- calculate-box-state-network-right
+  [state network]
+  (cond
+    (= state :error)
+    :error
+    (and (= network :arbitrum)
```

Comment:
could you explain how the network being arbitrum and state being "sending" means that it is confirmed?
I feel like I am not understanding the component

Reply:
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=22699-214190&mode=design&t=ENwvuByLXTCWyjRJ-0 Please check the design and the variants of the component here

Reply:
right, but why does this matter if it is arbitrum or optimism?

Reply:
as I understand it, this relates to this part of the design. I think if you were to swap values, e.g 
if the network values were swapped and it was confirmed on "Optimism" but pending on "Arbitrum" would it still be "state" pending? 

<img width="301" alt="Screenshot 2023-11-23 at 14 26 03" src="https://github.com/status-im/status-mobile/assets/22799766/0c1999fe-4bf8-4c02-bdb2-516f81749867">

Reply:
I guess you're right. Let's check with the design team.

Reply:
@J-Son89, @mmilad75, I think we should fully ignore Figma's state "pending". This state of the Figma design is really just to give the visual reference of how different component can be. 

In dev component we should configure all networks separately. No common state.
Instead of: 
```
{              :title                        "Title"
                :counter                      counter
                :total-box                    total-box
                :tag-name                     "Doodle"
                :tag-number                   "120"
                :epoch-number-mainnet         "181,329"
                :epoch-number-optimism        "181,329"
                :epoch-number-arbitrum        "181,329"
                :optimism-progress-percentage "10"
                :arbitrum-progress-percentage "10"
                :network                      :mainnet
                :state-mainnet                :pending
                :state-arbitrum               :pending
                :state-optimism               :pending
                :customization-color          :blue
                :tag-photo                    (resources/get-mock-image :collectible)
                :on-press                     (fn []
                                                (js/alert "Transaction progress item pressed"))}
```

We need:
```
{:title                        "Title"
 :counter                      counter
 :total-box                    total-box
 :tag-name                     "Doodle"
 :tag-number                   "120"
 :networks                       [{:optimism {:epoch-number "181,329"
                                           :progress "16"
                                           :state :pending}
                                :arbitrum {:epoch-number "181,329"
                                           :progress 48
                                           :state :error}]
                               }
 ...
 }

```

When  component needs to show only `:mainnet`, we pass only its config. When both - optimism or arbitrum - we path both configs within one property, as vector.
Our component should be dumb and configurable) No complicated internal logic, but possibility to show any state :)

Reply:
Sure, let's arrange one for tomorrow

Reply:
Done